workers/unity/Assets/Fps/Scripts/Editor/GameLogic/WorldTiles/MapQualitySettingsInspector.cs

```diff
@@ -0,0 +1,218 @@
+﻿using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+namespace Fps
+{
+    [CustomEditor(typeof(MapQualitySettings))]
+    public class MapQualitySettingsInspector : Editor
+    {
+        private int activeQualityLevel;
+        private MapQualitySettings settings;
+        private int StoredQualityLevel; // Used to restore project quality level when no longer previewing quality
+        private readonly Color dimColor = new Color(.77f, .77f, .77f);
+
+        private bool SettingsAreCurrentlyInUse =>
+            Application.isPlaying && MapQualitySettings.Instance == settings
+            || MapQualitySettings.ShowPreview;
+
+        private void OnEnable()
+        {
+            StoredQualityLevel = QualitySettings.GetQualityLevel();
+        }
+
+        private void OnDisable()
+        {
+            if (!MapQualitySettings.ShowPreview)
+            {
+                return;
+            }
+
+            MapQualitySettings.ShowPreview = false;
+            QualitySettings.SetQualityLevel(StoredQualityLevel);
+            settings.Apply();
+        }
+
+        public override void OnInspectorGUI()
+        {
+            settings = (MapQualitySettings) target;
+
+            activeQualityLevel = QualitySettings.GetQualityLevel();
+
+            DrawNotEditingActiveQualitySettingWarning();
+
+            EditorGUI.BeginChangeCheck();
+
+            RefreshQualityProperties();
+            DrawPropertiesExcluding(serializedObject,
+                serializedObject.FindProperty("Settings").name);
+
+            GUILayout.Space(10);
+
+            GUILayout.Label("Tile draw distances", EditorStyles.boldLabel);
+            DrawTileQualityProperties();
+
+            GUILayout.Space(10);
+
+            if (!Application.isPlaying
```

Comment:
This can just be `if(!Application.isPlayer || MapQualitySettings.Instance == settings)`

Reply:
And can wrap the next if statement as well.

`if((!Application.isPlayer || MapQualitySettings.Instance == settings) && DrawPreviewToggle())`